client/web/src/components/login-toggle.tsx

```diff
@@ -162,7 +162,12 @@ function LoginPopoverContent({
       </div>
       {!auth.canManageNode && (
         <>
-          {!auth.viewerIdentity ? (
+          {auth.serverMode === "readonly" ? (
+            <p className="text-gray-500 text-xs">
+              This web interface is running in read-only mode. Use the Tailscale
+              CLI or a different client app to manage this device.
```

Comment:
I like having this snippet here, although I fear it's a little confusing because of how general it is. I think we should leave it at "This web interface is running in read-only mode." and provide a link out to a KB that explains a little more based on platform.